nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -190,12 +223,53 @@ in {
         type = lib.types.port;
         default = 55002;
       };
+
       sidekiqThreads = lib.mkOption {
-        description = lib.mdDoc "Worker threads used by the mastodon-sidekiq service.";
+        description = lib.mdDoc "Worker threads used by the mastodon-sidekiq-* services. If `sidekiqProcesses` is configured and any processes specify null `threads`, this value is used.";
         type = lib.types.int;
         default = 25;
       };
 
+      sidekiqProcesses = lib.mkOption {
+        description = lib.mdDoc "How many Sidekiq processes should be used to handle background jobs, and which job classes they handle.";
+        type = with lib.types; attrsOf (submodule {
+          options = {
+            jobClasses = lib.mkOption {
+              type = listOf (enum [ "default" "push" "pull" "mailers" "scheduler" "ingress" ]);
+              description = lib.mdDoc "If not empty, which job classes should be executed by this process. *Only one process should handle the 'scheduler' class.*";
```

Comment:
Upstream seems to be really cautious about only running one `scheduler` queue, maybe we can just enforce this in an assertion?

Reply:
I really like the idea of doing this, but I'm not sure exactly how to go about it. Do you have any ideas off the top of your head? If not I'll play around with the repl a bit.

Reply:
Maybe something like this?

```nix
lib.count
  (x: x)
  (lib.mapAttrsToList
    (_: v: builtins.elem "scheduler" v.jobClasses || v.jobClasses == [] )
    { a.jobClasses = [ "scheduler" ]; b.jobClasses = [ "bla" ]; })
```

The result should probably be exactly one? Or just not larger than one?